cli/server.go

```diff
@@ -2157,6 +2157,10 @@ func ConnectToPostgres(ctx context.Context, logger slog.Logger, driver string, d
 
 	err = migrations.Up(sqlDB)
 	if err != nil {
+		// Checks for database schema version mismatch
+		if errors.Is(err, os.ErrNotExist) && strings.Contains(err.Error(), "no migration found for version") {
```

Comment:
`golang-migrate` does not provide a better way to handle these errors, it returns the error message as a string as seen here

https://github.com/golang-migrate/migrate/blob/c378583d782e026f472dff657bfd088bf2510038/migrate.go#L809